src/cmd/build.js

```diff
@@ -62,3 +69,35 @@ export function prepareBuildDir(buildDir: string): Promise {
     }))
     .then(() => buildDir);
 }
+
+
+/*
+ * Allows or ignores files when creating a ZIP archive.
+ */
+export class FileFilter {
+  filesToIgnore: Array<string>;
+
+  constructor({filesToIgnore}: Object = {}) {
+    this.filesToIgnore = filesToIgnore || [
```

Comment:
This PR looks great!

The only thing that I'm not completely sure about is related to the `filesToIgnore` defaults and how its overriding will work (especially when it will be available through a config file or a command line option), and as this feature is not exposed to the user yet, most of this could be discussed and resolved in follow-ups if we prefer.

here are a couple of doubts/ideas:

- by looking into the test cases I though that `manifest.json` is not a great example of possible `filesToIgnore` customization, because we probably want to be sure it is never excluded by mistake (e.g. by always prepending `"!manifest.json"` to this array of patterns) or the resulting xpi could be invalid even if the extension source dir was completely valid (excluding the hypotetic filesToIgnore customization).

- how about excluding any hidden file by default? (e.g. by using `"**/.*"` instead of `"**/.git"`

- if we suppose that the addon developer can customize the excluded/included patterns somehow, how about composing the default patterns and the user patterns instead of a providing a full override by default? e.g. something like `[].concat(mandatory_patterns, user_patterns, defaults_patterns)`